### PR TITLE
Refactor io plugins

### PR DESCRIPTION
**Pull request by s-weigand**
This PR address  some of the changes proposed in #586 

- [x] Splitup entry points to be more granular
- [x] Rename Io and possibly split it up
- [x] Make use of the `Io` instances and remove the fmt arg
- [x] Add a `get_dataloader` (maybe even `get_datawriter`) method and convenience function for `DataIoInterfaces`

Since each commit has a district purpose, it might be easiest to review this PR commit by commit.

The other changes will be done in a separate PR since this will simplify the review.
- [ ] Move all the plugin registration code in one module/package
- [ ] Add warning when a plugin replaces an existing one

**Testing**

Passing the tests is mandatory.

**Closing issues**

not yet

### CODE REVIEW

**Review by jsnel — APPROVED** [review]
More verbose yet more tidy. Looks good to me.
Needs approval from @joernweissenborn before merge.

**Review by joernweissenborn — APPROVED** [review]

**Pull request merged**
